app/src/main/java/de/danoeh/antennapod/activity/SelectSubscriptionActivity.java

```diff
@@ -0,0 +1,170 @@
+                getBitmapFromUrl(listItems.get(checkedPosition));
+            }
+        });
+
+    }
+
+    public List<Feed> getFeedItems(List<NavDrawerData.DrawerItem> items, List<Feed> result) {
+        for (NavDrawerData.DrawerItem item : items) {
+            if (item.type == NavDrawerData.DrawerItem.Type.TAG) {
+                getFeedItems(((NavDrawerData.TagDrawerItem) item).children, result);
+            } else {
+                Feed feed = ((NavDrawerData.FeedDrawerItem) item).feed;
+                if (!result.contains(feed)) {
+                    result.add(feed);
+                }
+            }
+        }
+        return result;
+    }
+
+    private void addShortcut(Feed feed, Bitmap bitmap) {
+        if (ShortcutManagerCompat.isRequestPinShortcutSupported(this)) {
+            Intent intent = new Intent(this, MainActivity.class);
+            intent.setAction(Intent.ACTION_MAIN);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            intent.putExtra(EXTRA_FEED_ID, feed.getId());
+            String id = "subscription-" + feed.getId();
+            IconCompat icon;
+
+            if (bitmap != null) {
+                icon = IconCompat.createWithAdaptiveBitmap(bitmap);
+            } else {
+                icon = IconCompat.createWithResource(this, R.drawable.ic_folder_shortcut);
+            }
+
+            ShortcutInfoCompat shortcut = new ShortcutInfoCompat.Builder(this, id)
+                    .setShortLabel(feed.getTitle())
+                    .setLongLabel(feed.getFeedTitle())
+                    .setIntent(intent)
+                    .setIcon(icon)
+                    .build();
+
+            ShortcutManagerCompat.requestPinShortcut(this, shortcut, null);
+        } else {
+            Toast.makeText(this, R.string.shortcut_unsupported_device,
+                    Toast.LENGTH_SHORT).show();
+        }
+
+    }
+
+    private void getBitmapFromUrl(Feed feed) {
+        Glide.with(this)
+                .asBitmap()
+                .load(feed.getImageUrl())
+                .listener(new RequestListener<Bitmap>() {
+                    @Override
+                    public boolean onLoadFailed(@Nullable GlideException e,
+                                                Object model,
+                                                Target<Bitmap> target,
+                                                boolean isFirstResource) {
+                        addShortcut(feed, null);
+                        return true;
+                    }
+
+                    @Override
+                    public boolean onResourceReady(Bitmap resource,
+                                                   Object model,
+                                                   Target<Bitmap> target,
+                                                   DataSource dataSource,
+                                                   boolean isFirstResource) {
```

Comment:
This takes a lot of vertical space (the method fills my whole screen even though it contains only a few actual statements). I would do something like this:

```suggestion
                    public boolean onResourceReady(Bitmap resource, Object model,
                            Target<Bitmap> target, DataSource dataSource, boolean isFirstResource) {
```